src/java/org/apache/cassandra/cql3/statements/schema/CreateTableStatement.java

```diff
@@ -402,7 +402,7 @@ private void fixupCompactTable(List<ColumnProperties> clusteringTypes,
         if (hasCounters)
             flags.add(TableMetadata.Flag.COUNTER);
 
-        boolean isStaticCompact = !isDense;
+        boolean isStaticCompact = !isDense && !isCompound;
```

Comment:
Because it has nothing to do with this patch, you just spot that as you were coding this ... if we want to fix that then create another ticket otherwise we would be changing completely irrelevant stuff ... where does that end? Just focus on we what need to do and create another ticket if necessary.